src/SayMore/UI/ComponentEditors/SessionBasicEditor.cs

```diff
@@ -619,5 +626,123 @@ private void HandlePeopleEditClick(object sender, EventArgs e)
+			{
+				var accessProtocolNode = sessionDoc.CreateElement("AccessProtocol");
+				accessProtocolNode.SetAttribute("Name", Program.CurrentProject.AccessProtocol);
+				XmlElement accessNode = sessionDoc.CreateElement("access");
+				accessNode.SetAttribute("type", "string");
+				accessNode.InnerText = choices[0].OptionName;
+				currentAccessCode = accessNode.InnerText;
+				accessProtocolNode.AppendChild(accessNode);
+				accessProtocolsNode.InsertAfter(accessProtocolNode, accessProtocolsNode.LastChild);
+				root.InsertAfter(accessProtocolsNode, root.LastChild);
+			}
+			else
+			{
+				var accessNode = isAccessProtocolExists.SelectSingleNode("access");
+				if (accessNode != null)
+				{
+					currentAccessCode = accessNode.InnerText;
+					var found = choices.Any(i => i.ToString() == accessNode.InnerText);
+					if (!found)
+					{
+						currentAccessCode = choices[0].OptionName;
+					}
+					accessNode.InnerText = currentAccessCode;
+				}
+			}
+			var accessInMetaFile = _file.MetaDataFieldValues.Find(v => v.FieldId == "access");
+			if (accessInMetaFile != null)
+			{
+				accessInMetaFile.Value = currentAccessCode;
+				_file.Save();
+			}
+
+			if (accessRootNode != null) accessRootNode.InnerText = currentAccessCode;
+
+			using (var sessionOutput = XmlWriter.Create(sessionFile, new XmlWriterSettings { Indent = true }))
+			{
+				sessionDoc.Save(sessionOutput);
+			}
+
+			return currentAccessCode;
+		}
+
+		/// ------------------------------------------------------------------------------------
+		/// <summary>
+		/// AccessProtocol and Access node get updated when the options are changed by the user on Dropdownlist
+		/// </summary>
+		/// ------------------------------------------------------------------------------------
+		private void _access_SelectedIndexChanged(object sender, EventArgs e)
+		{
+			if (!_isAccessOptionChanged || Program.CurrentProject == null) return;
+
+			var accessInMetaFile = _file.MetaDataFieldValues.Find(v => v.FieldId == "access");
+			if (accessInMetaFile == null || accessInMetaFile.Value.ToString() == _access.Text)
+				return;
+
+			string sessionFile = _file.PathToAnnotatedFile;
+			var choices = _accessOptions.ToList();
+			var sessionDoc = new XmlDocument();
+			using (var sessionReader = XmlReader.Create(sessionFile))
+			{
+				sessionDoc.Load(sessionReader);
+			}
+			XmlNode root = sessionDoc.DocumentElement;
+			if (root != null)
+			{
+				XmlNode accessRootNode = root.SelectSingleNode("access");
+				var isAccessProtocolExists = sessionDoc.SelectSingleNode("//AccessProtocols/AccessProtocol[@Name='" + Program.CurrentProject.AccessProtocol + "']");
+				if (isAccessProtocolExists != null)
+				{
+					string optionSelected = string.Empty;
+					var accessNode = isAccessProtocolExists.SelectSingleNode("access");
+					if (accessNode != null)
+					{
+						optionSelected = choices.FirstOrDefault(i => i.Description.ToString() == _access.Text)?.OptionName;
```

Comment:
Under what circumstances can the user select an access protocol that does not correspond to one of the options?